lib/SQL/Translator/Producer/MySQL.pm

```diff
@@ -457,11 +457,17 @@ sub create_table
     #
     my @constraint_defs;
     my @constraints = $table->get_constraints;
+
+    # Get the primary key, if there is one. No need to create an index for an
+    # FK that is also the PK. It may be a multi-field key, so generate a
+    # suitable string for comparison.
+    my $pk = join "\0", map $_->fields, grep $_->type eq PRIMARY_KEY, @constraints;
```

Comment:
Indeed. Looks like the code in master doesn't handle multi-column foreign keys, either. I'll update the PR incorporating your suggestion and fix that.